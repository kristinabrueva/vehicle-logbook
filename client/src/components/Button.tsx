import React from "react";
interface ButtonProps {
  id?: any;
  inline?: Boolean;
  text: String;
  type?: "button" | "submit" | "reset" | undefined;
  handleClick?: (e: any) => void;
}

const Button = ({
  id,
  inline,
  type,
  text,
  handleClick,
}: ButtonProps): JSX.Element => (
  <>
    <button
      type={type}
      className={`border-[1px] rounded-sm text-sm px-2 bg-gray-100 border-black py-1 max-w-fit ${
        inline ? "inline" : "block"
      }`}
      id={id}
      data-testid="testButton"
      onClick={handleClick}
    >
      {text}
    </button>
  </>
);

export default Button;
