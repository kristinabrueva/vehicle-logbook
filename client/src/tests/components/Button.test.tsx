import React from "react";
import { render } from "@testing-library/react";
import Button from "../../components/Button";

it("renders button correctly", () => {
  const { container, getAllByTestId, getAllByText } = render(
    <Button text="Click me" />
  );
  expect(container).toMatchSnapshot();
  expect(getAllByTestId("testButton")).toHaveLength(1);
  expect(getAllByText("Click me")).toHaveLength(1);
});
