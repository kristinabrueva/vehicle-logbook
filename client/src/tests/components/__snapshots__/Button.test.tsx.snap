// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<div>
  <button
    class="border-[1px] rounded-sm text-sm px-2 bg-gray-100 border-black py-1 max-w-fit block"
    data-testid="testButton"
  >
    Click me
  </button>
</div>
`;
